src/main/java/org/graylog/integrations/aws/transports/KinesisTransport.java

```diff
@@ -93,37 +92,63 @@ public void handleChangedThrottledState(boolean isThrottled) {
     public void doLaunch(MessageInput input) throws MisfireException {
 
         final Region region = Region.of(Objects.requireNonNull(configuration.getString(CK_AWS_REGION)));
-        final String assumeRoleArn = configuration.getString(AWSInput.CK_ASSUME_ROLE_ARN);
         final String key = configuration.getString(CK_ACCESS_KEY);
         final String secret = configuration.getString(CK_SECRET_KEY);
-        Preconditions.checkArgument(StringUtils.isNotBlank(key), "An AWS key is required.");
-        AwsCredentialsProvider awsCredentialsProvider = AWSService.buildCredentialProvider(key, secret);
-        Preconditions.checkArgument(StringUtils.isNotBlank(secret), "An AWS secret is required.");
-
-        // Assume role ARN functionality only applies to the Kinesis runtime (not to the setup flows).
-        if (StringUtils.isNotBlank(assumeRoleArn)) {
-            StsClient stsClient = StsClient.builder()
-                                           .region(region)
-                                           .credentialsProvider(awsCredentialsProvider).build();
-            String roleSessionName = String.format("API_KEY_%s@ACCOUNT_%s",
-                                                   key,
-                                                   stsClient.getCallerIdentity().account());
-            awsCredentialsProvider = StsAssumeRoleCredentialsProvider.builder()
-                                                                     .stsClient(stsClient)
-                                                                     .refreshRequest(r -> r.roleArn(assumeRoleArn)
-                                                                                           .roleSessionName(roleSessionName)).build();
-        }
-        this.kinesisConsumer = new KinesisConsumer(
-                nodeId, this, objectMapper, kinesisCallback(input), configuration.getString(CK_KINESIS_STREAM_NAME),
-                AWSMessageType.valueOf(configuration.getString(AWSCodec.CK_AWS_MESSAGE_TYPE)), region,
-                awsCredentialsProvider,
-                configuration.getInt(CK_KINESIS_RECORD_BATCH_SIZE, DEFAULT_BATCH_SIZE)
-        );
+        final String assumeRoleArn = configuration.getString(AWSInput.CK_ASSUME_ROLE_ARN);
+
+        final String dynamodbEndpoint = configuration.getString(AWSInput.CK_DYNAMODB_ENDPOINT);
```

Comment:
NB : Refactor
List<VPCEndpoints> lstOfVPCEndpoints =  configuration.awsService.getListOfVPCEndpoints() 
lstOfVPCEndpoints.forEach(x->validate(x))

Reply:
@danotorrey After doing some slow thinking, if this feature is an optional one, we can create a new enhancement, to address this changes in the future release. ec2 describe-vpc-endpoints -> describes all the vpc endpoints that is available in a region. What are the default VPC endpoints, that we are configuring , if no options are provided ?

Reply:
I really like this enhancement. Perhaps we can consider this for the future.